### PR TITLE
Modify the challenge submit page (form submission)

### PR DESCRIPTION
**Pull request by Sanji515**
#### Changes proposed in this pull request:

- Only form submission is included in this PR, docker based submission will be in the upcoming PR.

- Link to live demo: http://pr-147-evalai.surge.sh 

**<p align="center">Screenshot 1:</p>**

Screenshot from 2019-06-15 23-22-30

**<p align="center">Screenshot 2:</p>**

Screenshot from 2019-06-15 23-22-50

-

### CODE REVIEW

**Comment by Sanji515** [comment]
@lunayach @RishabhJain2018 @galipremsagar @Shekharrajak 

Please review.

File: src/app/components/challenge/challengesubmit/challengesubmit.component.html [hunk 1]
```diff
@@ -1,50 +1,149 @@
...
+                    [label]="'input_file'" [accept]="'.json, .zip, .txt, .tsv,
+                    .gz'" [placeholder]="''" [type]="'file'" [value]="''"
+                    #formsubmit></app-input>
+                </div>
+              </div>
+
+              <div class="row row-lr-margin">
+                <div class="align-left col-sm-10 col-lr-pad">
+                  <div class="input-field align-left">
+                    <app-input [label]="'method_name'" [placeholder]="'Method
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmit/challengesubmit.component.html]:
```suggestion
                    <app-input [label]="'method_name'" [placeholder]="'Method name (Optional)'"
```
That is, closing the string in the current line itself.

File: src/app/components/challenge/challengesubmit/challengesubmit.component.html [hunk 2]
```diff
@@ -1,50 +1,149 @@
...
+          <ul>
+            <b class="fw-bold content fs-14">Select phase:</b>
+            <app-selectphase [phases]="filteredPhases" [phaseSelectionType]="phaseSelectionType"
+              [phaseSelected]="phaseSelected()"></app-selectphase>
+
+            <div class="form">
+              <div class="row row-lr-margin">
+                <div class="col-md-10 col-sm-10 col-lr-pad">
+                  <app-input (change)="validateInput($event.target.value)"
+                    [label]="'input_file'" [accept]="'.json, .zip, .txt, .tsv,
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmit/challengesubmit.component.html]:
```suggestion
                    [label]="'input_file'" [accept]="'.json, .zip, .txt, .tsv, .gz'"
```
(Closing it in the same line)

File: src/app/components/challenge/challengesubmit/challengesubmit.component.ts [hunk 3]
```diff
@@ -186,7 +244,22 @@ export class ChallengesubmitComponent implements OnInit {
...
+        self.globalService.setFormValueForLabel(self.components, 'input_file', null);
+        self.globalService.setFormValueForLabel(self.components, 'method_name', '');
+        self.globalService.setFormValueForLabel(self.components, 'method_description', '');
+        self.globalService.setFormValueForLabel(self.components, 'project_url', '');
+        self.globalService.setFormValueForLabel(self.components, 'publication_url', '');
+      }
     );
   }
+
+  validateInput(inputValue) {
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmit/challengesubmit.component.ts:298]: `If-else` could be simplified to this:

```
    validateInput(inputValue) {
         this.inputFile = inputValue === null;
  }
```

File: src/app/components/utility/selectphase/selectphase.component.html [hunk 4]
```diff
@@ -1,21 +1,33 @@
-<div class="select-phase">
-  <div class="select-title heading">Select a phase:</div>
-  <div [class.inline]="publicRouter.url.indexOf('submissions') >= 0" class="phase-div" *ngFor="let item of phases">
-    <div class="card card-light" (click)="selectPhase(item)" [class.selected]="item['is_selected']">
+<span *ngIf="phaseSelectionType=='radioButton'">
+<li *ngFor="let item of phases">
```
Comment (lunayach) [inline src/app/components/utility/selectphase/selectphase.component.html:2]: `span` above is used only to bind `ngIf`. We could use `ng-container` instead, as `span` in vain is getting appended to DOM. (Reference, https://stackoverflow.com/questions/38585720/how-to-apply-multiple-template-bindings-on-one-element-in-angular/43846773#43846773)

Comment (Sanji515) [reply src/app/components/utility/selectphase/selectphase.component.html:2]: cool... thanks

File: src/app/services/challenge.service.ts [hunk 5]
```diff
@@ -326,15 +326,17 @@ export class ChallengeService {
    * @param phase  challenge phase id.
    * @param formData  Form Data for submission (file)
    */
-  challengeSubmission(challenge, phase, formData) {
+  challengeSubmission(challenge, phase, formData, callback = () => {}) {
```
Comment (lunayach) [inline src/app/services/challenge.service.ts:330]: Add `* @param callback  callback function` above.

File: src/app/services/global.service.ts [hunk 6]
```diff
@@ -305,6 +330,7 @@ export class GlobalService {
     components.map((item) => {
       if (item.label.toLowerCase() === label.toLowerCase()) {
         value = item;
+        console.log(value);
```
Comment (lunayach) [inline src/app/services/global.service.ts]: Remove this.

File: src/app/components/challenge/challengesubmit/challengesubmit.component.html [hunk 7]
```diff
@@ -1,50 +1,149 @@
...
+                <div class="align-left col-sm-10 col-lr-pad">
+                  <app-input [label]="'method_description'" [type]="'textarea'"
+                    [placeholder]="'Method description (Optional)'" #formsubmit></app-input><br>
+                </div>
+              </div>
+
+              <div class="row row-lr-margin">
+                <div class="align-left col-sm-10 col-lr-pad">
+                  <app-input [label]="'project_url'" [type]="'text'"
+                    [placeholder]="'Project URL (Optional)'" #formsubmit></app-input>
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmit/challengesubmit.component.html:74]: Should we have a Regex check for URL's?

**Comment by lunayach** [comment]
LGTM for functionality. Though there are some UI glitches here and there, but we can handle them separately in one PR, once PR's for all these components are merged.

File: src/app/components/challenge/challengesubmit/challengesubmit.component.html (hunk 7 above)
Comment (Sanji515) [reply src/app/components/challenge/challengesubmit/challengesubmit.component.html:74]: as there was no check in our current version so I haven't put there. @RishabhJain2018 what do you say about it?

Comment (RishabhJain2018) [reply src/app/components/challenge/challengesubmit/challengesubmit.component.html:74]: yes, a check for the URLs would be good.

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**